bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1049 @@
+                completable.completeExceptionally(new HttpUnauthorizedException("HTTP 2 request not authorized"));
+            } else {
+                completable.completeExceptionally(e);
+            }
+            fatalErrorDelayed(this, e);
+        }
+
+        /**
+         * Check the reply headers to see whether the request was authorised.
+         */
+        @Override
+        public void onHeaders(@Nullable Stream stream, @Nullable HeadersFrame frame) {
+            Objects.requireNonNull(frame);
+            MetaData metaData = frame.getMetaData();
+            if (metaData.isResponse()) {
+                Response responseMetaData = (Response) metaData;
+                int httpStatus = responseMetaData.getStatus();
+                switch (httpStatus) {
+                    case HttpStatus.UNAUTHORIZED_401:
+                    case HttpStatus.FORBIDDEN_403:
+                        handleHttp2Error(Http2Error.UNAUTHORIZED);
+                    default:
+                }
+                contentType = responseMetaData.getFields().get(HttpHeader.CONTENT_TYPE).toLowerCase();
+            }
+        }
+    }
+
+    /**
+     * Adapter for listening to regular HTTP 2 GET/PUT request stream events.
+     *
+     * It assembles the incoming text data into an HTTP 'content' entity. And when the last data frame arrives, it
+     * returns the full content by completing the CompletableFuture with that data.
+     *
+     * In addition to those handled by the parent, it handles the following fatal error events by notifying the
+     * containing class:
+     *
+     * <li>onIdleTimeout()</li>
+     * <li>onTimeout()</li>
+     */
+    private class ContentStreamListenerAdapter extends BaseStreamListenerAdapter<String> {
+        private final DataFrameCollector content = new DataFrameCollector();
+
+        @Override
+        public void onData(@Nullable Stream stream, @Nullable DataFrame frame, @Nullable Callback callback) {
+            Objects.requireNonNull(frame);
+            Objects.requireNonNull(callback);
+            synchronized (this) {
+                content.append(frame.getData());
+                if (frame.isEndStream() && !completable.isDone()) {
+                    completable.complete(content.contentAsString().trim());
+                    content.reset();
+                }
+            }
+            callback.succeeded();
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Stream stream, @Nullable Throwable x) {
+            handleHttp2Error(Http2Error.IDLE);
+            return true;
+        }
+
+        @Override
+        public void onTimeout(@Nullable Stream stream, @Nullable Throwable x) {
+            handleHttp2Error(Http2Error.TIMEOUT);
+        }
+    }
+
+    /**
+     * Class to collect incoming ByteBuffer data from HTTP 2 Data frames.
+     */
+    private static class DataFrameCollector {
```

Comment:
Since this file is rather large (> 1000 lines) consider extracting this class to another file.